design/open/pull-request.md

```diff
@@ -0,0 +1,28 @@
+# Pull-request Proposal
+
+Enable a way to submit changes to a branch by creating a request to merge changes.
+The pull-request captures the merge operation we request to apply.
+
+### Goals
+
+- Capture request to merge a branch or specific reference
+- Mechanism to review, discuss and approve changes before applying to a branch
+- Automation using actions can verify / provide feedback on pull-request
+
+### How it will work
+
+Introduce a new resource called pull-request at the repository level.
+The pull-request will hold owner (the user who created the PR), assignee, reviewers the source branch/reference and the target branch.
+Each reviewer can set approved/reject (or clear his response) on the PR.
+PR can be merge operation is enabled when at least one reviewer approves or when no reviewers are assigned.
+PR can be closed at any time.
+PR will not be updated when new commits into the source branch.
+Merge PR will perform a merge operation and merge the PR as merged. If a conflict or an error occurs while merging, it will be returned as a response to the request.
```

Comment:
That's tricky. I might be reviewing a PR that has no chances to be merged since there are conflicts.

Reply:
We can enable client side request to run dry merge - but currelty we don't know if a merge operation will have conflicts until we do the actual merge.

I think it rases another question - does a PR represent the merge itself or the request?
If we have a pending request to merge a commit or a branch, any future request to merge this PR should point to the same PR? Can we have multiple PRs requesting the same thing open in parallel? (will move this one to issues to discuss)